syncopy/tests/test_connectivity.py

```diff
@@ -0,0 +1,584 @@
+
+
+def run_padding_test(call, pad_length):
+    """
+    The callable should test a solution and support
+    a single keyword argument `pad_to_length`
+    """
+
+    pad_options = [pad_length, 'nextpow2', None]
+    for pad in pad_options:
+        call(pad_to_length=pad)
+
+    # test invalid pads
+    try:
+        call(pad_to_length=2)
+    except SPYValueError as err:
+        assert 'pad_to_length' in str(err)
+        assert 'expected value to be greater' in str(err)
+
+    try:
+        call(pad_to_length='IamNoPad')
+    except SPYValueError as err:
+        assert 'Invalid value of `pad_to_length`' in str(err)
+        assert 'nextpow2' in str(err)
+
+    try:
+        call(pad_to_length=np.array([1000]))
+    except SPYValueError as err:
+        assert 'Invalid value of `pad_to_length`' in str(err)
+        assert 'nextpow2' in str(err)
+
+
+def run_cfg_test(call, method, positivity=True):
+
+    cfg = get_defaults(connectivity)
+
+    cfg.method = method
+    cfg.foilim = [0, 70]
+    cfg.taper = 'parzen'
+    cfg.output = 'abs'
+
+    result = call(cfg)
+
+    # check here just for finiteness and positivity
+    assert np.all(np.isfinite(result.data))
+    if positivity:
+        assert np.all(result.data[0, ...] >= -1e-10)
+
+
+def run_foi_test(call, foilim, positivity=True):
+
+    # only positive frequencies
+    assert np.min(foilim) >= 0
+    assert np.max(foilim) <= 500
+
+    # fois
+    foi1 = np.arange(foilim[0], foilim[1]) # 1Hz steps
+    foi2 = np.arange(foilim[0], foilim[1], 0.25) # 0.5Hz steps
+    foi3 = 'all'
+    fois = [foi1, foi2, foi3, None]
+
+    for foi in fois:
+        # FIXME: this works for method='granger' but not method='coh' 0.0
+
+        result = call(foi=foi, foilim=None)
+        # check here just for finiteness and positivity
+        assert np.all(np.isfinite(result.data))
+        if positivity:
+            assert np.all(result.data[0, ...] >= -1e-10)
+
+    # 2 foilims
+    foilims = [[2, 60], [7.65, 45.1234], None]
+    for foil in foilims:
+        result = call(foilim=foil, foi=None)
+        # check here just for finiteness and positivity
+        assert np.all(np.isfinite(result.data))
+        if positivity:        
+            assert np.all(result.data[0, ...] >= -1e-10)
+
+    # make sure specification of both foi and foilim triggers a
+    # Syncopy ValueError
+    try:
+        result = call(foi=foi, foilim=foil)
+    except SPYValueError as err:
+        assert 'foi/foilim' in str(err)
+
+    # make sure out-of-range foi selections are detected
+    try:
+        result = call(foilim=[-1, 70], foi=None)
+    except SPYValueError as err:
+        assert 'foilim' in str(err)
+        assert 'bounded by' in str(err)
+
+    try:
+        result = call(foi=np.arange(550, 700), foilim=None)
+    except SPYValueError as err:
+        assert 'foi' in str(err)
+        assert 'bounded by' in str(err)
+
+
+def mk_selection_comb(nTrials, nChannels, toi_min, toi_max):
+
```

Comment:
This makes a lot of sense - so much so, that we might want to migrate this to `tests/misc.py` down the line (where `generate_artificial_data` and friends live). What do you think?

Reply:
Renamed to `mk_selection_dicts`, originally I wanted to return the generators, but this way it's more flexible.